examples/allocation-endpoint/README.md

```diff
@@ -0,0 +1,99 @@
+
+In this folder you will find:
+
+1. Terraform modules that create resources in GCP
+2. The Allocation Endpoint Proxy code that gets deployed to Cloud Run
+3. An [ESP container](https://cloud.google.com/endpoints/docs/grpc/get-started-cloud-run) that gets deployed as a sidecar alongside the `agones-allocator`
+4. A sample client code to send allocation requests to the proxy
+5. Dockerfile that builds the proxy and scripts to push the image to a docker repository
+6. Documentation on how to use the solution.
+
+Here is the architecture of GCP resources created:
+
+![architecture](https://github.com/googleforgames/agones/blob/main/examples/allocation-endpoint/architecture.png?raw=true)
+
+
+## GKE cluster
+First and foremost you need to create clusters and install Agones to experiment with this solution.
+The clusters can be in the same GCP project as your Allocation Endpoint proxy or they can be in a different project.
+
+---
+**NOTE**
+
+The solution has not been tested with non-GKE clusters.
+
+---
+
+When creating GKE Clusters, [Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) needs to be enabled. Please follow [GKE cluster setup](https://agones.dev/site/docs/installation/creating-cluster/gke/) and include workload-pool, e.g:
+
+<pre>
+gcloud container clusters create [NAME] \
+  ...
+  <b>--workload-pool=[GKE-PROJECT-ID].svc.id.goog \ </b>
+</pre>
+
+Install Agones on your cluster. You need to disable mTLS because the `agones-allocator` container will be the backend for ESP container.
+
+```
+helm upgrade my-release --install --namespace agones-system --create-namespace agones/agones \
+  --set agones.allocator.disableMTLS=true \
+  --set agones.allocator.disableTLS=true \
+  --set agones.allocator.service.http.enabled=false
+```
+
+After installing Agones, deploy [ESP](https://cloud.google.com/endpoints/docs/grpc/specify-esp-v2-startup-options) which is an envoy based proxy in front of agones `agones-alloator` container. Run the following to patch the service deployement, change the service port to ESP and add annotation to `agones-allocator` service account to impersonate GCP service account. 
+
+Replace [GKE-PROJECT-ID] in `patch-agones-allocator.yaml` with your project ID before runing the scripts.
+
+```
+kubectl patch deployment agones-allocator -n agones-system --patch-file patch-agones-allocator.yaml
+kubectl patch svc agones-allocator -n agones-system --type merge -p '{"spec":{"ports": [{"port": 443,"name":"https","targetPort":9443}]}}'
+kubectl annotate sa -n agones-system agones-allocator iam.gke.io/gcp-service-account=ae-esp-sa@[PROJECT-ID].iam.gserviceaccount.com
+```
+
+## Terraform 
+The terraform modules create resources in GCP:
+
+```
+terraform apply -var "project_id=[PROJECT-ID]" -var "ae_proxy_image=[GCR-IMAGE]:[VERSION]" -var "authorized_members=[\"serviceAccount:[SERVICE-ACCOUNT-EMAIL]\"]" -var "clusters_info=[CLUSTERS-INFO]" -var "workload-pool=[GKE-PROJECT-ID].svc.id.goog"
```

Comment:
You can replace `-var "ae_proxy_image=[GCR-IMAGE]:[VERSION]"` with `-var "ae_proxy_image=gcr.io/agones-images:0.1"` and make a note that you can change it to your own image if you want to customize the proxy code. 

Alternatively, since this is already set as the default image value in variables.tf you could shorten this line by not specifying it at all here.

Reply:
Update - I prefer leaving out the `ae_proxy_image` altogether to make the walkthrough simpler.

Reply:
Removed it.